provisioning/Samples/service/ProvisioningAzureSasCredentialSample/Program.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Azure;
+using CommandLine;
+using Microsoft.Azure.Devices.Provisioning.Service;
+using System;
+using System.Globalization;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using System.Timers;
+
+namespace ProvisioningAzureSasCredentialSample
+{
+    internal class Program
+    {
+        private static TimeSpan timeToLive = TimeSpan.FromHours(1);
+        private static string sasToken;
+        private static AzureSasCredential sasCredential;
+        private static Parameters parameters = null;
+
+        public static async Task Main(string[] args)
+        {
+            // Parse application parameters
+            ParserResult<Parameters> parserResult = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            // Initialize SAS token credentials
+            Console.WriteLine("Creating SAS credential.");
+
+            DateTime expiresOn = DateTime.UtcNow.Add(timeToLive);
+            sasToken = GenerateSasToken(parameters.HostName, parameters.SharedAccessKey, parameters.SharedAccessKeyName, expiresOn);
+            // Note: Pass the generated sasToken and not just the shared access signature when creating the AzureSasCredential.
+            sasCredential = new AzureSasCredential(sasToken);
+
+            // This is an example of how to create a Timer that will automatically update the SAS credential every hour, to avoid expiry
+            // Since this sample is short, this timer event will not trigger, and thus this section can be ignored/removed
+            Timer updateTimer = new Timer();
+            updateTimer.Interval = 3600000;
+            updateTimer.Elapsed += UpdateToken;
+            updateTimer.AutoReset = true;
+            updateTimer.Enabled = true;
```

Comment:
Will this work?
```suggestion
            var updateTimer = new Timer
            {
                Interval = 60 * 60 * 1000, // 1 hour in milliseconds
                Elapsed = UpdateToken,
                AutoReset = true,
                Enabled = true,
            };
```

Reply:
No, Elapsed can only be on the left hand side of a += or -= operation, which aren't allowed in initializers